alternator/executor.cc

```diff
@@ -2951,11 +2951,38 @@ static future<executor::request_return_type> do_query(schema_ptr schema,
     });
 }
 
+static dht::token token_for_segment(int segment, int total_segments) {
+    assert(total_segments > 1 && segment < total_segments);
```

Comment:
Maybe also assert segment >= 0 if this is also a requirement?
In any case I guess none of these asserts are critical, because you test these things right before calling this code.

Reply:
Yes, I can add one more, it makes sense. The only semi-critical assertion I originally put here was `total_segments > 0`, since I don't want to end the world by trying to divide by zero :) The other conditions were added later.